crates/wasmparser/src/lib.rs

```diff
@@ -27,6 +27,28 @@
 //! the examples documented for [`Parser::parse`] or [`Parser::parse_all`].
 
 #![deny(missing_docs)]
+#![no_std]
+
+extern crate alloc;
+#[cfg(feature = "std")]
+extern crate std;
```

Comment:
I'd suggest going with what has been described in this comment: https://github.com/rust-lang/rust/issues/121708#issuecomment-1974964139

I did the refactoring in Wasmi and the code reads a lot better as we can use `std` for nearly all imports. Also the warnings mentioned in this issue are no longer a deal.

Reply:
Sorry I'm a bit confused, but isn't that what's happening here?

Reply:
Yeah sorry, the approach I actually wanted to suggest was to use:
```rust
#![no_std]

#[cfg(not(feature = "std"))]
extern crate alloc as std;

#[cfg(feature = "std")]
extern crate std;
```
So that you can keep most of the `std` imports which reduces the PR diff. Also works with the warnings issue that the linked thread is about.

Reply:
fair enough!